src/Bicep.Core/TypeSystem/TypeHelper.cs

```diff
@@ -27,7 +27,9 @@ public static class TypeHelper
             {
                 if (unionType.Members.All(x => x is StringLiteralType || x == LanguageConstants.String))
                 {
-                    return LanguageConstants.String;
+                    return unionType.Members.Any(x => x == LanguageConstants.String) ?
+                        LanguageConstants.String :
+                        unionType;
```

Comment:
This would still collapse open enums down to strings, right? That may be moot since there's no userland way to declare one, but this could return `unionType` in all cases if that type information should be preserved.